pkg/route/api/validation/validation.go

```diff
@@ -76,6 +79,64 @@ func ValidateRouteStatusUpdate(route *routeapi.Route, older *routeapi.Route) fie
 	return allErrs
 }
 
+// ValidateRouteTLSConfig checks if the TLS config (certificates and keys) for a route is valid.
+func ValidateRouteTLSConfig(route *routeapi.Route) field.ErrorList {
+	tls := route.Spec.TLS
+	result := field.ErrorList{}
+
+	tlsFieldPath := field.NewPath("spec").Child("tls")
+	if errs := validateTLS(route, tlsFieldPath); len(errs) != 0 {
+		result = append(result, errs...)
+	}
+
+	// TODO: Check if we can be stricter with validating the certificate
+	//       is for the route hostname. Don't want existing routes to
+	//       break, so disable the hostname validation for now.
+	// hostname := route.Spec.Host
+	hostname := ""
+	certPool := x509.NewCertPool()
+
+	if len(tls.CACertificate) > 0 {
+		cacert, err := validateCertificatePEM(tls.CACertificate, nil)
```

Comment:
Why would we want to add them to the pool? We just want to check that the CA Certificate given to us is valid and the chain is valid and the leaf is signed by a well-known authority.